examples/demo_bgp.pp

```diff
@@ -130,81 +130,148 @@
-    capability_negotiation => true,
-    dynamic_capability     => true,
-    ebgp_multihop          => 2,
-    local_as               => 55.77,
-    log_neighbor_changes   => disable,
-    low_memory_exempt      => false,
-    remote_as              => 12,
-    remove_private_as      => 'all',
-    shutdown               => true,
-    suppress_4_byte_as     => true,
-    timers_keepalive       => 90,
-    timers_holdtime        => 270,
-    update_source          => 'ethernet1/1',
-    transport_passive_only => false,
-  }
+  if $operatingsystem == 'nexus' {
+    cisco_bgp_neighbor {'55.77 blue 1.1.1.1':
+      ensure                 => present,
+
+      #Properties
+      description            => 'my description',
+      connected_check        => true,
+      capability_negotiation => true,
+      dynamic_capability     => true,
+      ebgp_multihop          => 2,
+      local_as               => 55.77,
+      log_neighbor_changes   => disable,
+      low_memory_exempt      => false,
+      remote_as              => 12,
+      remove_private_as      => 'all',
+      shutdown               => true,
+      suppress_4_byte_as     => true,
+      timers_keepalive       => 90,
+      timers_holdtime        => 270,
+      update_source          => 'ethernet1/1',
+      transport_passive_only => false,
+    }
+
+    cisco_bgp_neighbor {'55.77 blue 2.2.2.2':
+      ensure                 => present,
 
-  cisco_bgp_neighbor {'55.77 blue 2.2.2.2':
-    ensure                 => present,
-
-    #Properties
-    description            => 'my description',
-    connected_check        => true,
-    capability_negotiation => true,
-    dynamic_capability     => true,
-    ebgp_multihop          => 2,
-    log_neighbor_changes   => disable,
-    low_memory_exempt      => false,
-    remote_as              => 12,
-    remove_private_as      => 'all',
-    shutdown               => true,
-    suppress_4_byte_as     => true,
-    timers_keepalive       => 90,
-    timers_holdtime        => 270,
-    update_source          => 'ethernet1/1',
-    transport_passive_only => false,
+      #Properties
+      description            => 'my description',
+      connected_check        => true,
+      capability_negotiation => true,
+      dynamic_capability     => true,
+      ebgp_multihop          => 2,
+      log_neighbor_changes   => disable,
+      low_memory_exempt      => false,
+      remote_as              => 12,
+      remove_private_as      => 'all',
+      shutdown               => true,
+      suppress_4_byte_as     => true,
+      timers_keepalive       => 90,
+      timers_holdtime        => 270,
+      update_source          => 'ethernet1/1',
+      transport_passive_only => false,
+    }
+
+    cisco_bgp_neighbor {'55.77 blue 3.3.3.3':
+      ensure      => present,
+
+      #Properties
+      description => 'config_for_rr_client',
+      remote_as   => 3604557,
+      shutdown    => false,
+    }
   }
+  elsif $operatingsystem == 'ios_xr' {
```

Comment:
It would be worth looking into the various [conditional operators](http://docs.puppetlabs.com/puppet/4.3/reference/lang_conditional.html#selectors) to see if you can have OS-dependent logic and shared type declaration, something like:

```puppet
$capability_negotiation = $operatingsystem ? {
  'nexus' => true
  default => undef
}
$dynamic_capability = $operatingsystem ? {
  'nexus' => true
  default => undef
}

    # actual neighbor declaration is shared between both OS:
    cisco_bgp_neighbor {'55.77 blue 1.1.1.1':
      ensure                 => present,

      #Properties
      description            => 'my description',
      connected_check        => true,
      capability_negotiation => $capability_negotiation,
      dynamic_capability     => $dynamic_capability,
      ebgp_multihop          => 2,
      local_as               => 55.77,
...
```